sysdig: 0.35.3 -> 0.36.0

SebTM opened the pull request:
## Description of changes

Updated sysdig to latest release v0.36.0 and fixed issue with 6.8-kernel (https://github.com/draios/sysdig/issues/2077)
Fixed file-formatting
Mark broken packages as broken

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

SebTM commented:
Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>linuxKernel.packages.linux_4_19_hardened.sysdig</li>
    <li>linuxKernel.packages.linux_5_4_hardened.sysdig</li>
    <li>linuxKernel.packages.linux_lqx.sysdig</li>
  </ul>
</details>
<details>
  <summary>19 packages built:</summary>
  <ul>
    <li>linuxKernel.packages.linux_4_19.sysdig</li>
    <li>linuxKernel.packages.linux_5_10.sysdig</li>
    <li>linuxKernel.packages.linux_5_10_hardened.sysdig</li>
    <li>linuxKernel.packages.linux_5_15.sysdig</li>
    <li>linuxKernel.packages.linux_5_15_hardened.sysdig</li>
    <li>linuxKernel.packages.linux_5_4.sysdig</li>
    <li>linuxKernel.packages.linux_6_1.sysdig</li>
    <li>linuxKernel.packages.linux_6_1_hardened.sysdig</li>
    <li>linuxKernel.packages.linux_6_6.sysdig</li>
    <li>linuxKernel.packages.linux_hardened.sysdig (linuxKernel.packages.linux_6_6_hardened.sysdig)</li>
    <li>linuxKernel.packages.linux_6_7.sysdig</li>
    <li>linuxKernel.packages.linux_6_7_hardened.sysdig</li>
    <li>linuxKernel.packages.linux_6_8.sysdig</li>
    <li>linuxKernel.packages.linux_latest_libre.sysdig</li>
    <li>linuxKernel.packages.linux_libre.sysdig</li>
    <li>linuxKernel.packages.linux_xanmod.sysdig</li>
    <li>linuxKernel.packages.linux_xanmod_latest.sysdig (linuxKernel.packages.linux_xanmod_stable.sysdig)</li>
    <li>linuxKernel.packages.linux_zen.sysdig</li>
    <li>sysdig</li>
  </ul>
</details>

SebTM commented on pkgs/os-specific/linux/sysdig/default.nix:
```diff
@@ -107,39 +114,37 @@ stdenv.mkDerivation rec {
-    broken = kernel != null && versionOlder kernel.version "4.14";
+    maintainers = [ maintainers.raskin ];
+    platforms = [ "x86_64-linux" ] ++ platforms.darwin;
+    broken = kernel != null && ((versionOlder kernel.version "4.14") || (kernel.isHardened && lib.lists.findSingle (x: x == kernel.version) false true [ "4.19" "5.4" "lqx" ]));
```
Is that working like I expect it/how can I ensure it the build output seems not to change (nixpkgs-review) if I mark it as broken?

jfly replied:
Sorry, I'm not familiar with marking packages broken or how it interacts with nixpkgs-review.

nit, feel free to ignore: IMO, this does seem involved enough that I'd appreciate a comment (or a link to an issue) describing the way(s) this build is broken (and ideally issues filed in the relevant upstream place(s) if there's any hope of fixing this).

SebTM replied:
Marking them as broken comes from the last PR https://github.com/NixOS/nixpkgs/pull/293688#issuecomment-1987811365

It appears to me e.g. in case of 4.19 (only different options but same error for all 3 hardened-kernels) that some functions got stripped out (to harden) this causes 

`/nix/store/pi6cvclrd4ca1g6z525fsa71f4ayj8n7-linux-config-4.19.309.drv`

to fail with e.g.
```
warning: unused option: CFI_CLANG
warning: unused option: CONFIG_LDISC_AUTOLOAD
error: unused option: GCC_PLUGINS
error: unused option: GCC_PLUGIN_LATENT_ENTROPY
error: unused option: GCC_PLUGIN_RANDSTRUCT
error: unused option: GCC_PLUGIN_RANDSTRUCT_PERFORMANCE
warning: unused option: GCC_PLUGIN_STRUCTLEAK
warning: unused option: GCC_PLUGIN_STRUCTLEAK_BYREF_ALL
warning: unused option: INET_DIAG_DESTROY
warning: unused option: INET_MPTCP_DIAG
warning: unused option: INET_RAW_DIAG
warning: unused option: INET_TCP_DIAG
warning: unused option: INET_UDP_DIAG
warning: unused option: IOMMU_DEFAULT_DMA_LAZY
warning: unused option: IOMMU_DEFAULT_DMA_STRICT
warning: unused option: IO_STRICT_DEVMEM
warning: unused option: SECURITY_WRITABLE_HOOKS
warning: unused option: SLS
warning: unused option: STRICT_DEVMEM
warning: unused option: UBSAN_LOCAL_BOUNDS
```

SO I think sysdig is just not supported on these hardend-kernels due to their hardening. (assumption)

SebTM commented on pkgs/os-specific/linux/sysdig/default.nix:
```diff
@@ -72,6 +72,13 @@ stdenv.mkDerivation rec {
     cp -r ${driver} driver-src
     chmod -R +w driver-src
+
+    # Hacky but needed until https://github.com/draios/sysdig/issues/2077 is resolved for kernel >= 6.8 as strlcpy got removed and build fails
```
This is hacky but the simplest solution (I found) to fix the build and due to the condition in the code I'm sure it's not executed for 6.8-kernel - is there a better way?

Tried combinations with "-Wno-error" etc. but seems to not be help.

SebTM replied:
The issue is this is a build error breaking the build so I can't ignore it without patching it...

jfly reviewed (commented):
Could you split the formatting changes into a separate commit? That would make it easier to see the material changes in this PR.

SebTM committed:
sysdig: 0.35.3 -> 0.36.0

SebTM committed:
sysdig: fix formattig

SebTM commented:
> Could you split the formatting changes into a separate commit? That would make it easier to see the material changes in this PR.

Done :)

jfly reviewed (approved):
LGTM, but I am not a maintainer of this package.

SebTM commented on pkgs/os-specific/linux/sysdig/default.nix:
```diff
@@ -4,35 +4,34 @@
   };
 
-  # https://github.com/draios/sysdig/blob/0.35.1/cmake/modules/driver.cmake
+  # https://github.com/draios/sysdig/blob/0.36.0/cmake/modules/driver.cmake
```
It's easier for me to lookup the new-release and also update the comments because I need to open these specific links for the release I'm doing shouldn't matter/was there before I started helping out on sysdig :)

SebTM commented on pkgs/os-specific/linux/sysdig/default.nix:
```diff
@@ -72,6 +71,13 @@ stdenv.mkDerivation rec {
+
+    # Hacky but needed until https://github.com/draios/sysdig/issues/2077 is resolved for kernel >= 6.8 as strlcpy got removed and build fails
+    ${lib.optionalString
+    (kernel != null && lib.versionAtLeast kernel.version "6.8") ''
```
> I'd guess it's safe to do in all kernel versions we support.
I wouldn't do this it's only removed from kernel >= 6.8 in the kernel-sources build against and so only causing an issue there as the function definition is missing even it would never be used because it has a check in the pm_events.c for kernel-version.

Mic92 committed:
sysdig: disable hardened and zen kernels

Also it currently compiles with zen kernels, it doesn't with lqx.
Since we currently don't have a mechanism of lqx only, we can re-add,
if one of the lqx/zen maintainers steps up and also help maintaining and testing sysdig.

Mic92 commented on pkgs/os-specific/linux/sysdig/default.nix:
```diff
@@ -107,39 +114,37 @@ stdenv.mkDerivation rec {
-    broken = kernel != null && versionOlder kernel.version "4.14";
+    maintainers = [ maintainers.raskin ];
+    platforms = [ "x86_64-linux" ] ++ platforms.darwin;
+    broken = kernel != null && ((versionOlder kernel.version "4.14") || kernel.isHardened || kernel.isZen);
```
Disabled non-standard kernel configuration for now, since they cause too much trouble.
If someone feels strongly about supporting them, I would expect them also to step up and help maintaining them.
